engine_test.go

```diff
@@ -15,53 +15,73 @@
 package engine
 
 import (
+	"context"
+	"errors"
+	"path/filepath"
 	"testing"
-)
 
-var TestUnitType UnitType = 999
+	"github.com/stretchr/testify/assert"
+)
 
-type TestRule struct{}
+type successTestRule struct{}
 
-func (rule TestRule) IsFor(unitType UnitType) bool {
-	return TestUnitType == unitType
+func (rule *successTestRule) Run(_ string) ([]Finding, error) {
+	return nil, nil
 }
 
-type TestUnit struct{}
+type failTestRule struct{}
 
-func (unit TestUnit) Type() UnitType {
-	return TestUnitType
+func (rule *failTestRule) Run(_ string) ([]Finding, error) {
+	return nil, errors.New("some error")
 }
 
-func (unit TestUnit) Eval(rule Rule) []Finding {
-	return []Finding{
-		Finding{
-			ID: "1",
+func TestEngineRun(t *testing.T) {
+	testcases := []struct {
+		name                    string
+		extensions              []string
+		expectedVulnerabilities int
+		rules                   Rule
+		projectPath             string
+		err                     bool
+	}{
+		{
+			name:                    "Should run and detect vulnerabilities with no error",
+			extensions:              []string{".go"},
+			rules:                   &successTestRule{},
+			expectedVulnerabilities: 0,
+			projectPath:             filepath.Join(".", "text", "examples"),
+			err:                     false,
+		},
+		{
+			name:                    "Should have error when invalid path",
+			extensions:              []string{".go"},
+			expectedVulnerabilities: 0,
+			rules:                   &successTestRule{},
+			projectPath:             "invalidPath",
+			err:                     true,
+		},
+		{
+			name:                    "Should have error when rules run error",
```

Comment:
Shouldn't be "... when rules return an error"?

Reply:
Done!